h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -1261,8 +1270,8 @@ public void computeImpl() {
         updateProgress(false);
       // lambda search loop
       for (int i = 0; i < _parms._lambda.length; ++i) {  // lambda search
-        if(_parms._max_iterations != -1 && _state._iter >= _parms._max_iterations)
-          break;
+        if (_job.stop_requested() || (timeout() && _model._output._submodels.length > 0)) break;  //need at least one submodel on timeout to avoid issues.
+        if(_parms._max_iterations != -1 && _state._iter >= _parms._max_iterations) break;
```

Comment:
@wendycwong , can you please verify that those interruptions can't do any harm?
Ideally, I'd like for GLM to stop gracefully in case of timeout, possibly with a not-fully trained model.
In case of stop user request, we raise an exception, so I'm less concerned...

Reply:
I feel this should work.  As long as your fix pass Jenkin GLM tests, I feel confident.

Reply:
This part looks fine.